sktime/transformations/compose.py

```diff
@@ -663,3 +664,137 @@ def _underscore_join(iterable):
         """Create flattened column names from multiindex tuple."""
         iterable_as_str = [str(x) for x in iterable]
         return "__".join(iterable_as_str)
+
+
+class FitInTransform(BaseTransformer):
+    """Transformer compostion to always fit a given transformer on the transform data only.
+
+    Some transformers like Imputer(method="mean") might be more accurate for some use
```

Comment:
I would recommend to be more precise here, along the following lines. It´s not about "accuracy", but you may want to apply the transformation to the same sample. This can make sense in classification, but usually doesn´t make sense in forecasting (since it can lead to information leakage).

Reply:
pls provide description you want to have

Reply:
done